src/panels/ActionsPanel/ActionOptions.tsx

```diff
@@ -0,0 +1,53 @@
+import { useMemo } from "react";
+import { Formik } from "formik";
+
+import OptionInputs from "components/RadioInputBox/OptionInputs";
+
+import type {
+  ActionData,
+  ActionOptionsType,
+  OnValuesChange,
+} from "panels/ActionsPanel/ActionsPanel";
+
+type Props = {
+  name: string;
+  data: ActionData;
+  onValuesChange: OnValuesChange;
+};
+
+export default function ActionOptions({ name, data, onValuesChange }: Props) {
+  const action = data[name];
+
+  const collectedOptions = useMemo(() => {
+    const collectOptions: ActionOptionsType = [];
+    Object.keys(action.params.properties).forEach((name) => {
+      const property = action.params.properties[name];
+      const required = action.params.required;
+      collectOptions.push({
+        name: name,
+        description: property.description,
+        type: property.type,
+        required: required ? required.includes(name) : false,
+      });
+    });
+    return collectOptions;
+  }, [action.params.properties, action.params.required]);
```

Comment:
Could this just depend on `action.params` or even `action`?

Reply:
technically yes, but `params` has more fields so this felt a bit safer wrt unnecessary re-renders.